python/dlisio/ext/core.cpp

```diff
@@ -371,26 +371,28 @@ noexcept (false) {
                 // const auto msg = 'Non-sequential frames. expected = {}, current = {}'
             }
 
-            const auto tail = std::distance(ptr, end);
-            auto size_src = pre_size.src + data_size.src + post_size.src;
-            if (tail < size_src) {
-                const auto msg = "unaligned record: tail (which is "
-                               + std::to_string(tail)
-                               + ") < fmt_size (which is "
-                               + std::to_string(size_src)
-                               + ")"
-                               ;
-                throw std::runtime_error(msg);
-            }
+            auto assert_overflow = [end](const char* ptr, int skip) {
+                if (ptr + skip > end) {
+                    const auto msg = "corrupted record: fmtstr would read past end";
+                    throw std::runtime_error(msg);
+                }
+            };
 
-            ptr += pre_size.src;
```

Comment:
It seems to me like you don't need pre_size and post_size variables anymore.
If I understand correctly, even if there are varsize data in pre-fmt or post-fmt, it shouldn't be a problem anymore.

We don't have tests dedicated for this, but you can add the following to test_fdata:
```

def test_channel_curves():
    fpath = 'data/chap4-7/iflr/all-reprcodes.dlis'
    with dlisio.load(fpath) as (f, *_):
        channel = f.object('CHANNEL', 'CH26', 10, 0)
        curves = channel.curves()
        np.testing.assert_array_equal(
            curves[0], np.array(True))
```

There are some ascii etc data before and after channel

Reply:
We don't _right now_, but in the very near future, fixed-size records will get a different code path, and they will come back.